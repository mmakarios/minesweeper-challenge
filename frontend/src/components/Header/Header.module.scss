@import "../../styles/variables";

.header {
  background-color: var(--background);
  display: flex;
  padding: 1rem 2rem;
  position: fixed;
  right: 0;

  > span {
    font-size: 2rem;
  }
}
