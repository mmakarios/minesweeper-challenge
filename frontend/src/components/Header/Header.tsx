import { useState, useEffect, useCallback } from "react";
import ToggleSwitch from "../ToggleSwitch";
import styles from "./Header.module.scss";

interface Props {}

export const Header = (props: Props) => {
  const [isDarkTheme, setIsDarkTheme] = useState<boolean>(true);
  useEffect(() => {
    if (isDarkTheme) {
      window.document.documentElement.dataset.theme = "theme-dark";
      return;
    }
    window.document.documentElement.dataset.theme = "theme-light";
  }, [isDarkTheme]);

  const onToggleTheme = useCallback(() => {
    setIsDarkTheme(!isDarkTheme);
  }, [isDarkTheme, setIsDarkTheme]);

  return (
    <header className={styles.header}>
      <span>{isDarkTheme ? "🌑" : "☀️"}</span>
      <ToggleSwitch checked={isDarkTheme} onClick={onToggleTheme} />
    </header>
  );
};

export default Header;
