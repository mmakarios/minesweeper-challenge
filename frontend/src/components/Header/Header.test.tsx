import React from "react";
import renderer, { act } from "react-test-renderer";
import Header from "./Header";

it("renders component", () => {
  const tree = renderer.create(<Header />);
  const label = tree.root.findByType("label");
  act(() => {
    label.props.onClick();
    tree.update(<></>);
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
