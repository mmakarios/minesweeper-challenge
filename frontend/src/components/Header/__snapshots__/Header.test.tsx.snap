// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<header
  className="header"
>
  <span>
    ðŸŒ‘
  </span>
  <div
    className="toggleSwitch"
  >
    <input
      checked={true}
      type="checkbox"
    />
    <label
      onClick={[Function]}
    />
  </div>
</header>
`;
